### DISCUSSION TITLE
Bagaimana Cara Memanggil Asset JS melalui Inline JS?

### DISCUSSION THREAD

**Discussion by yuliakbard** [mecha-cms/mecha#260] [Q&A]
Kang, Bagaimana cara memanggil asset JS melalui inline JS?

misalnya ane, ingin menambahkan service worker di web ane. dengan inline script dasar nya seperti ini:

``` js
if ('serviceWorker' in navigator) {
    navigator.serviceWorker.register('sw.js');
}
```

ane ingin memanggil asset `sw.js` yang tersimpan di asset layout.

**Comment by taufik-nurrohman** [comment]
Buat apa itu mas? Bikin aplikasi obrol?

~~~ php
$script = <<<JS
if ('serviceWorker' in navigator) {
    navigator.serviceWorker.register('sw.js');
}
JS;

Asset::set('data:text/js;base64,' . To::base64($script), 10);
~~~

**Reply by yuliakbard** [reply]
gak kang, mau tambahin script biar installable aja.

**Reply by yuliakbard** [reply]
![image](https://user-images.githubusercontent.com/96925342/221341940-699d0b63-31dd-4bf4-942a-a5da8ba75a3d.png)

ini kenapa ya kang? error jadinya

**Reply by yuliakbard** [reply]
ini log error

![image](https://user-images.githubusercontent.com/96925342/221341976-66d6b428-75ab-467b-bb89-50a776f8d484.png)

ini log error-y

![image](https://user-images.githubusercontent.com/96925342/221341984-2ac0d4ec-71ee-42fc-b2b5-d371dec0e8ad.png)

**Reply by taufik-nurrohman** [reply]
Metode `Asset::script()` sudah tak hilangkan karena nggak ada cara untuk bisa melepasnya setelah terpasang. Sekarang pakai model URL Base64 seperti yang Saya tuliskan. Pakai `Asset::set()` sudah cukup.

**Reply by taufik-nurrohman** [reply]
> ![image](https://user-images.githubusercontent.com/96925342/221341940-699d0b63-31dd-4bf4-942a-a5da8ba75a3d.png)
> 
> ini kenapa ya kang? error jadinya

Kalau respons berubah jadi begini biasanya karena berkas `index.php` di layout berubah nama jadi `.index.php` karena sebelumnya terjadi _error_.
